lib/dfu/src/main/java/no/nordicsemi/android/dfu/DfuBaseService.java

```diff
@@ -1035,7 +1035,11 @@ public void onCreate() {
 
 		final LocalBroadcastManager manager = LocalBroadcastManager.getInstance(this);
 		final IntentFilter actionFilter = makeDfuActionIntentFilter();
-		manager.registerReceiver(mDfuActionReceiver, actionFilter);
+		if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+			this.registerReceiver(mDfuActionReceiver, actionFilter, RECEIVER_NOT_EXPORTED);
+		} else {
+			manager.registerReceiver(mDfuActionReceiver, actionFilter);
+		}
 		// Additionally we must register this receiver as a non-local to get broadcasts from the notification actions
 		ContextCompat.registerReceiver(this, mDfuActionReceiver, actionFilter, ContextCompat.RECEIVER_NOT_EXPORTED);
```

Comment:
Now the OS receiver is registered twice, as this line does the same as 1039. We're using `ContextCompat` here to avoid the need for branching based on version number.